fl4health/clients/deep_mmd_clients/ditto_deep_mmd_client.py

```diff
@@ -0,0 +1,201 @@
+from logging import ERROR
+from pathlib import Path
+from typing import Dict, Optional, Sequence, Tuple
+
+import torch
+import torch.nn as nn
+from flwr.common.logger import log
+from flwr.common.typing import Config, Scalar
+
+from fl4health.checkpointing.client_module import CheckpointMode, ClientCheckpointModule
+from fl4health.clients.basic_client import TorchInputType
+from fl4health.clients.ditto_client import DittoClient
+from fl4health.losses.deep_mmd_loss import DeepMmdLoss
+from fl4health.model_bases.feature_extractor_buffer import FeatureExtractorBuffer
+from fl4health.utils.losses import EvaluationLosses, LossMeterType
+from fl4health.utils.metrics import Metric
+
+
+class DittoDeepMmdClient(DittoClient):
+    def __init__(
+        self,
+        data_path: Path,
+        metrics: Sequence[Metric],
+        device: torch.device,
+        loss_meter_type: LossMeterType = LossMeterType.AVERAGE,
+        checkpointer: Optional[ClientCheckpointModule] = None,
+        lam: float = 1.0,
+        deep_mmd_loss_weight: float = 10.0,
+        flatten_feature_extraction_layers: Dict[str, bool] = {},
+        size_feature_extraction_layers: Dict[str, int] = {},
+    ) -> None:
+        """
+        This client implements the DEEP-MMD loss function in the Ditto framework.
+        Args:
+            data_path (Path): path to the data to be used to load the data for client-side training
+            metrics (Sequence[Metric]): Metrics to be computed based on the labels and predictions of the client model
+            device (torch.device): Device indicator for where to send the model, batches, labels etc. Often 'cpu' or
+                'cuda'
+            loss_meter_type (LossMeterType, optional): Type of meter used to track and compute the losses over
+                each batch. Defaults to LossMeterType.AVERAGE.
+            checkpointer (Optional[TorchCheckpointer], optional): Checkpointer to be used for client-side
+                checkpointing. Defaults to None.
+            metrics_reporter (Optional[MetricsReporter], optional): A metrics reporter instance to record the metrics
```

Comment:
I think this is probably an issue with the DittoClient docstring as well, but the `metrics_reporter` is technically not an arg in this implementation.